File: mycollab-services/src/main/java/com/esofthead/mycollab/module/crm/service/ibatis/ContactOpportunityServiceImpl.java
/**
 * This file is part of mycollab-services.
 *
 * mycollab-services is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-services is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-services.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.esofthead.mycollab.module.crm.service.ibatis;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.esofthead.mycollab.core.persistence.ISearchableDAO;
import com.esofthead.mycollab.core.persistence.service.DefaultSearchService;
import com.esofthead.mycollab.module.crm.dao.ContactOpportunityMapperExt;
import com.esofthead.mycollab.module.crm.domain.criteria.ContactSearchCriteria;
import com.esofthead.mycollab.module.crm.service.ContactOpportunityService;

/**
 * 
 * @author MyCollab Ltd.
 * @since 3.0
 * 
 */
@Service
public class ContactOpportunityServiceImpl extends
		DefaultSearchService<ContactSearchCriteria> implements
		ContactOpportunityService {

	@Autowired
	private ContactOpportunityMapperExt contactOpportunityMapperExt;

	@Override
	public ISearchableDAO<ContactSearchCriteria> getSearchMapper() {
		return contactOpportunityMapperExt;
	}

}

File: mycollab-services/src/main/java/com/mycollab/module/crm/service/impl/ContactOpportunityServiceImpl.kt
package com.mycollab.module.crm.service.impl

import com.mycollab.db.persistence.ISearchableDAO
import com.mycollab.db.persistence.service.DefaultSearchService
import com.mycollab.module.crm.dao.ContactOpportunityMapperExt
import com.mycollab.module.crm.domain.criteria.ContactSearchCriteria
import com.mycollab.module.crm.service.ContactOpportunityService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

/**
 * @author MyCollab Ltd.
 * @since 3.0
 */
@Service
class ContactOpportunityServiceImpl : DefaultSearchService<ContactSearchCriteria>(), ContactOpportunityService {

    @Autowired
    private val contactOpportunityMapperExt: ContactOpportunityMapperExt? = null

    override val searchMapper: ISearchableDAO<ContactSearchCriteria>
        get() = contactOpportunityMapperExt as ISearchableDAO<ContactSearchCriteria>

}
